services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -427,3 +428,84 @@ async def is_dynamic_service_running(
         )
 
         return len(dynamic_sidecar_services) == 1
+
+
+async def get_or_create_networks_ids(
+    networks: List[str], project_id: ProjectID
+) -> Dict[str, str]:
+    async def _get_id_from_name(client, network_name: str) -> str:
+        network = await client.networks.get(network_name)
+        network_inspect = await network.show()
+        return network_inspect["Id"]
+
+    async with docker_client() as client:
+        existing_networks_names = {x["Name"] for x in await client.networks.list()}
+        log.debug("existing_networks_names=%s", existing_networks_names)
+
+        # create networks if missing
+        for network in networks:
+            if network not in existing_networks_names:
+                network_config = {
+                    "Name": network,
+                    "Driver": "overlay",
+                    "Labels": {
+                        "com.simcore.description": "study service communication network",
+                        # TODO: use below when closing project and GC to remove network
+                        "project_id": f"{project_id}",
+                    },
+                    "Attachable": True,
+                    "Internal": True,  # no internet access
+                }
+                try:
+                    await client.networks.create(network_config)
+                except aiodocker.exceptions.DockerError:
+                    # multiple calls to this function can be processed in parallel
+                    # this will cause creation to fail, it is OK to assume it already
+                    # exist an raise an error (see below)
+                    log.info(
+                        "Network %s might already exist, skipping creation", network
+                    )
+
+        ids = await logged_gather(
+            *[_get_id_from_name(client, network) for network in networks]
+        )
+
+        if len(ids) != len(networks):
```

Comment:
why not checking if the results of logged_gather return any exception instead?

Reply:
Since logged_gather already raises when it finishes its election, it is not required to check it.

Reply:
then I do not quite understand the need to check the len?

Reply:
Good point. After a more careful look, I've removed it. Made no sense.